src/Database/Schema/PostgresSchemaDialect.php

```diff
@@ -609,12 +609,39 @@ public function columnSql(TableSchema $schema, string $name): string
     {
         $data = $schema->getColumn($name);
         assert($data !== null);
+        $data['name'] = $name;
 
         $sql = $this->_getTypeSpecificColumnSql($data['type'], $schema, $name);
         if ($sql !== null) {
             return $sql;
         }
+        $autoIncrementTypes = [
+            TableSchemaInterface::TYPE_TINYINTEGER,
+            TableSchemaInterface::TYPE_SMALLINTEGER,
+            TableSchemaInterface::TYPE_INTEGER,
+            TableSchemaInterface::TYPE_BIGINTEGER,
+        ];
+        $primaryKey = $schema->getPrimaryKey();
+        if (
+            in_array($data['type'], $autoIncrementTypes, true) &&
+            $primaryKey === [$name]
+        ) {
+            $data['autoIncrement'] = true;
```

Comment:
This drops the `id` name requirement.

Reply:
I was trying to align how postgres works with other database platforms which don't require the column to be called `id`.

Reply:
Feels like a mistake to force every integer key to be auto-increment.

Reply:
In any case, I don't think we can make that change in 5.x.

Reply:
Fair point, I'll add the column name check back in.